draft/affordances.md

```diff
@@ -0,0 +1,56 @@
+---
+RFC: XXXX
+Author: Kyle Fuller
+Status: Draft
+Created: 2015-09-21
+Last Modified: 2015-09-21
+---
+
+# API Blueprint RFC XXXX: Affordances
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Rationale](#rationale)
+    - [Example](#example)
+
+## Abstract
+
+This RFC proposes that you can define affordances (link relations) that an API
+resource or action provides.
+
+## Rationale
+
+This RFC proposes adding an `Affordances` section into API Blueprint that
+sits under a resource or action providing the affodances that it provides.
+
+### Example
+
+To describe affordances from a `Question` resource to two other resources, we
+can describe them within the affordances section as follows:
+
+```apib
+## Question [/question/{id}]
+
++ Parameters
+    + id: 1
+
++ Affordances
+    + `create_comment` (Comments[create])
+    + `author` (User)
```

Comment:
@danielgtaylor AFAIK, for example, this is trying to define the `_links` keyword in HAL. But that raises a lot more questions for me.

Let's take the following example on the HAL spec,

```json
{
    "_links": {
        "self": { "href": "/orders" },
        "curies": [{ "name": "ea", "href": "http://example.com/docs/rels/{rel}", "templated": true }],
        "next": { "href": "/orders?page=2" },
        "ea:find": {
            "href": "/orders{?id}",
            "templated": true
        },
        "ea:admin": [{
            "href": "/admins/2",
            "title": "Fred"
        }, {
            "href": "/admins/5",
            "title": "Kate"
        }]
    },
}
```

As daniel said above, we really don't know the relations between the `Question` resource and `User` resource. Similarly with `Comment` resource. We can use `templated` as shown above, but I think some people might prefer the actual values instead of templates. Also, If you look at the above example, how do we define something like the `ea:admin`?

There has been a bit of work done by @zdne on [Resource Blueprint](https://github.com/apiaryio/api-blueprint/tree/resource-blueprint#def-use-case-api) which tries to solve these issues and talks about state machines.

In my opinion, the introduction of `Affordances` to API Blueprint as it is in this document is an half-attempt at the resource blueprint which I don't think does anything good for us. It might actually hamper the development in the future.